src/neo-vm/ExecutionEngine.cs

```diff
@@ -1074,12 +1071,11 @@ private bool ExecuteInstruction()
             return true;
         }
 
-        private void LoadContext(ExecutionContext context)
+        protected virtual void LoadContext(ExecutionContext context)
```

Comment:
Will we still inherit from this?

Reply:
Is the unique way to set the states

Reply:
So event based may be better than this, if we can seal the class and use event to manage internals...